File: ingwang/programmers-2/의상.java
import java.util.HashMap;

class Solution27 {
    public static int solution27(String[][] clothes) {
        int answer = 1;
        HashMap<String , Integer> hashmap = new HashMap<>();
        int num = 0;
        for (int i = 0; i < clothes.length; i++) {
            if(hashmap.containsKey(clothes[i][1])){
                num = hashmap.get(clothes[i][1]);
                hashmap.put(clothes[i][1], num+1);
            }else{
                hashmap.put(clothes[i][1], 1);
            }
        }

        for (String key : hashmap.keySet()){
            answer *= hashmap.get(key)+1;
        }
        answer-=1;
        return answer;
    }

    public static void main(String[] args) {
        String[][] a = new String[][]{{"yellow_hat", "headgear"},{"blue_sunglasses", "eyewear"},{"green_turban", "headgear"}};
        System.out.println(solution27(a));
    }
}

File: sanghoon/programmers-2/의상.py
from collections import Counter
def solution(clothes):

    clothes_count = {}

    for item in clothes :                       # 의상 종류 별 갯수 딕셔너리 화
        item_type = item[1]
        clothes_count[item_type] = clothes_count.get(item_type,0) + 1
    answer = 1
    for i in clothes_count.values() :           # 의상 가지수에 +1해서 거듭제곱
        answer *= (i+1)
    print(answer-1)
    return answer - 1                           # 아무것도 입지 않은 경우 -1
clothes = [["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]]
solution(clothes)